### PR TITLE
:gift: E2E tests

### LINKED ISSUE #78: [QE] Implement e2e tests for sending events to public and cluster-local resources
The e2e tests should be written that will send events to public and cluster-local resources.

Tests should be executed as part of integration test pass on pre-submit.

The kn-event already has some integration tests, scattered thought the source code. They are skipped if no Kube connection is provided. Those also should be executed, in integration tests pass.

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- :gift: E2E tests structure
- :gift: Executing package level integration tests within E2E run
- :gift: Testing in-cluster-send to k8s service, kn service, broker, and channel

/kind enhancement

Fixes #78

### CODE REVIEW

**Comment by cardil** [comment]
What is left to do (mind dump before holiday season):

1. The test sometimes fails to send the event in first go. The waiting for successful job is faulty and in the end only successful at first time passes (see: #130). While running on GKE, the error can be seen only on Google Logs Explorer. The error is as follows. This should not happen, as sink is reported to be ready (adding readiness probe to it, doesn't help). The solution would be to fix #130, but that may hide the possible flaw in reconciler-test framework. Adding about 5sec wait also help mitigate this.
   ```
   "can't send with ICS sender: can't sent event: Post "http://sink-mcmtbmdo.test-zweizver.svc.cluster.local": dial tcp 10.147.251.7:80: connect: connection refused"
   ```
2. Additional test cases should be added. Current one deploys regular K8s service. We should add cases for Knative service (we could reuse [_wathola-forwarder image_](https://github.com/knative/eventing/tree/main/test/test_images/wathola-forwarder) to forward events to sink), Broker, and Channel.

**Comment by cardil** [comment]
/hold

Still missing the broker and channel test case for in-cluster sender test

**Review by jrangelramos — COMMENTED** [review]
In general it looks good. I didn't go too deep on the framework but took a look and ran it.

I would suggest to add a little more comments in some files (e.g. `ics_send.go`)
Also to get it runnig I had to hack a bit the linters such as disabling `nolintlint` and removing `ireturn`, and `varnamelen` from disable list.

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+function knative_setup() {
+  start_latest_knative_serving
+  start_latest_knative_eventing
+}
+
+initialize "$@"
+
+set -Eeuo pipefail
+
+./mage publish
```
[the remark this answers is not in the document]
Comment (cardil) [reply test/e2e-tests.sh:29]: We need to build and publish sender image. We'll need some solution to overwrite it on downstream projects, but I leave it for future work in follow up PR

[the remark this answers is not in the document]
Comment (cardil) [reply test/e2e-tests.sh:29]: > Also to get it runnig I had to hack a bit the linters such as disabling `nolintlint` and removing `ireturn`, and `varnamelen` from disable list.

I bet, you've got an old `golangci-lint`. Consider upgrade.

**Review by dsimansk — APPROVED** [review]
Thanks!

/approve
/lgtm

**Comment by cardil** [comment]
/unhold

I'll add channel and broker tests in follow up PR.

**Pull request merged**